### DISCUSSION TITLE
Using styled() with raw html elements

### DISCUSSION THREAD

**Discussion by apperside** [tamagui/tamagui#254] [Ideas]
It would be useful to be able to use `styled()` with native html elements in order to be able to using style props wherever you need it.
In my specific case I am trying to render a table, and I wanted to style my `table`, `tr`, `td` elements, and I tried the following:

```typescript
const MyTable = styled(Stack, { tag: 'table'})
const MyTh = styled(Stack, { tag: 'th' })
const MyTd = styled(Stack, { tag: 'td' })
```

But those elements (don't know why) do not render like their native counterpart (for example, if I use `<td>`  it behaves different than if I use <MyTd>.

The api for this feature could be similar to the one used by `styled-components`:

```typescript
const MyTd = styled("td");
```

**Comment by nandorojo** [comment]
I don't think the API should support `styled('td')`, as this wouldn't work on native. If there's an issue with the `tag` prop, that should just get fixed instead.
